lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/math/VectorUtils.java

```diff
@@ -69,6 +69,69 @@ public static Vector3d nearestPointOnLine(Vector3d point, Vector3d linePt1, Vect
         return result;
     }
 
+    /**
+     * Enum to differentiate between orientations of a point relative to a line segment.
+     */
+    private enum Orientation2d {
+        COLLINEAR, CLOCKWISE, COUNTERCLOCKWISE
+    }
+
+    /**
+     * Function to calculate the orientation of the triplet (px, py), (qx, qy), (rx, ry).
+     */
+    private static Orientation2d isCollinear(double px, double py, double qx, double qy, double rx, double ry) {
+        double val = (qy - py) * (rx - qx) - (qx - px) * (ry - qy);
+        if (Math.abs(val) <= MathUtils.EPSILON_D) { // Collinear
+            return Orientation2d.COLLINEAR;
+        }
+        return (val > 0) ? Orientation2d.CLOCKWISE : Orientation2d.COUNTERCLOCKWISE; // Clockwise or Counterclockwise
+    }
+
+    /**
+     * Function to check if point (qx, qy) lies on a segment (px, py) to (rx, ry).
+     */
+    private static boolean isOnSegment(double px, double py, double qx, double qy, double rx, double ry) {
+        return qx <= Math.max(px, rx)
+                && qx >= Math.min(px, rx)
+                && qy <= Math.max(py, ry)
+                && qy >= Math.min(py, ry);
+    }
+
+    /**
+     * Function to check if two line segments (p1, q1) and (p2, q2) intersect.
+     *
+     * @param p1 First point of first line segment
+     * @param q1 Second point of first line segment
+     * @param p2 First point of second line segment
+     * @param q2 Second point of second line segment
+     * @return {@code true} if the two line segments intersect, else {@code false}
+     */
+    public static boolean doesXZIntersect(Vector3d p1, Vector3d q1, Vector3d p2, Vector3d q2) {
```

Comment:
can you rename them to a1 b1 and a2 and b2? p and q are so similar it's utterly confusing to read.

Reply:
and also I would move `isCollinear `and `isOnSegment` below `doesXZIntersect`